src/Data/Bin/Ordering.agda

```diff
@@ -0,0 +1,790 @@
+  where
+  open ℕₚ.≤-Reasoning
+  xN      = toℕ x
+  yN      = toℕ y
+  xN<yN   = toℕ-mono-< x<y
+  x'N<y'N = begin
+    ℕ.suc (ℕ.suc (2 ℕ.* xN))   ≡⟨ refl ⟩
+    2 ℕ.+ (2 ℕ.* xN)            ≡⟨ sym (ℕₚ.*-distribˡ-+ 2 1 xN) ⟩
+    2 ℕ.* (ℕ.suc xN)            ≤⟨ ℕₚ.*-monoʳ-≤ 2 xN<yN ⟩
+    2 ℕ.* yN                     ≤⟨ ℕₚ.*-monoʳ-≤ 2 (ℕₚ.≤-step ℕₚ.≤-refl) ⟩
+    2 ℕ.* (ℕ.suc yN)            ∎
+
+
+toℕ-mono-< {1+[2 x ]} {2[1+ .x ]} (odd<even (inj₂ refl)) =  x'N<y'N
+  where
+  xN = toℕ x
+
+  x'N<y'N :  ℕ.suc (ℕ.suc (2 ℕ.* xN))  ℕ.≤  2 ℕ.* (ℕ.suc xN)
+  x'N<y'N =  ℕₚ.≤-reflexive (sym (ℕₚ.*-distribˡ-+ 2 1 xN))
+
+toℕ-mono-< {1+[2 x ]} {1+[2 y ]} (odd<odd x<y) =  x'N<y'N
+  where
+  xN      = toℕ x
+  yN      = toℕ y
+  xN<yN   = toℕ-mono-< x<y
+  x'N<y'N = ℕₚ.+-monoʳ-< 1 (ℕₚ.*-monoʳ-< 1 xN<yN)
+
+
+fromℕ-mono-< :  fromℕ Preserves ℕ._<_ ⟶ _<_
+fromℕ-mono-< {m} {n} m<n =  aux (<-cmp x y)
+  where
+  open P.≡-Reasoning
+  x       = fromℕ m;      y       = fromℕ n
+  toℕ-x≡m = toℕ-fromℕ m;  toℕ-y≡n = toℕ-fromℕ n
+
+  aux : Tri (x < y) (x ≡ y) (x > y) → x < y
+  aux (tri< x<y _   _) =  x<y
+  aux (tri≈ _   x≡y _) =  contradiction m≡n (ℕₚ.<⇒≢ m<n)
+    where
+    m≡n = begin
+      m       ≡⟨ sym toℕ-x≡m ⟩
+      toℕ x   ≡⟨ cong toℕ x≡y ⟩
+      toℕ y   ≡⟨ toℕ-y≡n ⟩
+      n       ∎
+  aux (tri> _ _ y<x) =  contradiction m<n (ℕₚ.<⇒≯ n<m)
+    where
+    yN<xN = toℕ-mono-< y<x;  n<m = subst₂ ℕ._<_ toℕ-y≡n toℕ-x≡m yN<xN
+
+
+fromℕ-mono-≤ :  fromℕ Preserves ℕ._≤_ ⟶ _≤_
+fromℕ-mono-≤ {m} {n} m≤n  with ℕₚ.m≤n⇒m<n∨m≡n m≤n
+... | inj₁ m<n =  inj₁ (fromℕ-mono-< m<n)
+... | inj₂ m≡n =  inj₂ (cong fromℕ m≡n)
+
+toℕ-mono-≤ :  toℕ Preserves _≤_ ⟶ ℕ._≤_
+toℕ-mono-≤ {x} {y} (inj₁ x<y) =  ℕₚ.<⇒≤ (toℕ-mono-< x<y)
+toℕ-mono-≤ {x} {y} (inj₂ x≡y) =  ℕₚ.≤-reflexive (cong toℕ x≡y)
+
+------------------------------------------------------------------------------
+-- Reflexivity, transitivity, antisymmetry
+------------------------------------------------------------------------------
+
+≤-refl :  Reflexive _≤_
+≤-refl =  inj₂ refl
+
+≤-reflexive : _≡_ ⇒ _≤_
+≤-reflexive {x} {_} refl =  ≤-refl {x}
+
+≤-trans : Transitive _≤_
+≤-trans (inj₁ x<y)  (inj₁ y<z)             =  inj₁ (<-trans x<y y<z)
+≤-trans {x} {_} {_} (inj₁ x<y)  (inj₂ y≡z) =  inj₁ (subst (x <_) y≡z x<y)
+≤-trans {_} {_} {z} (inj₂ x≡y) (inj₁ y<z)  =  inj₁ (subst (_< z) (sym x≡y) y<z)
+≤-trans             (inj₂ x≡y) (inj₂ y≡z)  =  inj₂ (trans x≡y y≡z)
+
+≤-antisym : Antisymmetric _≡_ _≤_
+≤-antisym (inj₂ x≡y) _          =  x≡y
+≤-antisym _          (inj₂ y≡x) =  sym y≡x
+≤-antisym (inj₁ x<y) (inj₁ y<x) =  contradiction y<x (<⇒≯ x<y)
+
+------------------------------------------------------------------------------
+-- Several other properties for _<_, _≤_.
+------------------------------------------------------------------------------
+
+≤→≢→< :  ∀ {x y} → x ≤ y → x ≢ y → x < y
+≤→≢→< (inj₁ x<y) _   =  x<y
+≤→≢→< (inj₂ x≡y) x≢y =  contradiction x≡y x≢y
+
+0≤ :  ∀ x → zero ≤ x
```

Comment:
Please could you add `x`s to all these names,

Reply:
Done.